generator/src/test/java/com/scottlogic/deg/generator/restrictions/DateTimeRestrictionsTests.java

```diff
@@ -0,0 +1,247 @@
+        restrictions.max = new DateTimeRestrictions.DateTimeLimit(LocalDateTime.MAX, false);
+
+        boolean result = restrictions.match(LocalDateTime.MAX);
+
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void matchShouldReturnFalseIfGivenDateIsBeforeMin(){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        LocalDateTime limit = LocalDateTime.of(2001, 02, 03, 04, 05, 06);
+        restrictions.min = new DateTimeRestrictions.DateTimeLimit(limit, false);
+
+        boolean result = restrictions.match(limit.minusSeconds(1));
+
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void matchShouldReturnFalseIfGivenDateIsAfterMax(){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        LocalDateTime limit = LocalDateTime.of(2001, 02, 03, 04, 05, 06);
+        restrictions.max = new DateTimeRestrictions.DateTimeLimit(limit, false);
+
+        boolean result = restrictions.match(limit.plusSeconds(1));
+
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    public void matchShouldReturnTrueIfGivenDateIsAfterMinExclusiveAndBeforeMaxExclusive(){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        LocalDateTime min = LocalDateTime.of(2001, 02, 03, 04, 05, 06);
+        LocalDateTime max = min.plusHours(1);
+        restrictions.min = new DateTimeRestrictions.DateTimeLimit(min, false);
+        restrictions.max = new DateTimeRestrictions.DateTimeLimit(max, false);
+
+        boolean result = restrictions.match(min.plusMinutes(1));
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    public void matchShouldReturnTrueIfGivenDateIsEqualToMinAndMaxInclusive(){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        LocalDateTime min = LocalDateTime.of(2001, 02, 03, 04, 05, 06);
+        restrictions.min = new DateTimeRestrictions.DateTimeLimit(min, true);
+        restrictions.max = new DateTimeRestrictions.DateTimeLimit(min, true);
+
+        boolean result = restrictions.match(min);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    public void matchShouldReturnFalseIfGivenDateIsEqualToMinAndMaxExclusive(){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        LocalDateTime min = LocalDateTime.of(2001, 02, 03, 04, 05, 06);
+        restrictions.min = new DateTimeRestrictions.DateTimeLimit(min, false);
+        restrictions.max = new DateTimeRestrictions.DateTimeLimit(min, false);
+
+        boolean result = restrictions.match(min);
+
+        Assert.assertFalse(result);
+    }
+
+    private DateTimeRestrictions restrictions(MockDateTimeLimit min, MockDateTimeLimit max){
+        DateTimeRestrictions restrictions = new DateTimeRestrictions();
+        restrictions.min = min;
+        restrictions.max = max;
+        return restrictions;
+    }
+
+    private class MockDateTimeLimit extends DateTimeRestrictions.DateTimeLimit{
```

Comment:
Can we use a mocking framework?

Reply:
Unfortunately not, mockito doesn't support asserting/verifying `equals` or `hashCode`
https://github.com/mockito/mockito/wiki/FAQ